bin/waagent

```diff
@@ -24,9 +24,13 @@
 #
 
 import os
-import importlib
 import sys
 
+if sys.version_info[0] == 2:
+    import imp
+else:
+    import importlib
```

Comment:
does importlib do anything or potentially going to do anything here?

Reply:
`importlib` is the preferred module since `imp` is deprecated. We define it here, the same way like `imp` was defined here so far. Does that answer your question?

Reply:
Sorry, i understand that imp is deprecated for python3, and importlib is the succeeder. But i don't see any usage of importlib in this file, and the code snippet where was using 'imp' is never supposed to be executed with python3, so i think it would be the same if we just remove the else part. Please correct me if i'm wrong, thanks!

Reply:
@rainer37 I see your point now. You are right, since we have:
```
    if sys.version_info[0] == 3:
        raise ImportError("waagent2.0 doesn't support python3")
```
`imp` would never be executed with python3.

However, the imports will always be evaluated, which means that today, in an image that runs the agent with Python3, we would see this:
```
root@ubuntu2004:~# waagent --version
/usr/sbin/waagent:27: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  import imp
WALinuxAgent-2.2.46 running on ubuntu 20.04
Python: 3.8.2
Goal state agent: 2.2.46
```
The purpose of this change is to get rid of this warning.

Reply:
@pgombar thanks for the info! but sorry i think 
```
if sys.version_info[0] == 2:
    import imp
```
is enough to silence the DeprecationWarning for os with /usr/bin/env python as python3* since python3 wouldn't take this branch, and python3 is not going to do any module load using 'importlib'. S adding
```
else:
    import importlib
```
does not seem to help in any cases but only import a module that is not used. Any further explanation would be much appreciated, thanks!

Reply:
Looking only at this file's contents, you are right and there is no need for `importlib` since it is not called anywhere in this copy of `bin/waagent`. However, there are certain implicit dependencies between `bin/waagent` and VM extensions. In case the extensions need to load modules from the agent, and are using Python3, we are enabling them to do so with `importlib`.